x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -260,6 +268,15 @@ private void gatherMetrics(LogicalPlan plan, BitSet b) {
      * Limit QL's comparisons to types we support.
      */
     public static Failure validateBinaryComparison(BinaryComparison bc) {
+        return validateBinaryComparison((BinaryScalarFunction) bc);
+    }
+
+    public static Failure validateBinaryComparison(InsensitiveBinaryComparison bc) {
+        return validateBinaryComparison((BinaryScalarFunction) bc);
+    }
+
+    private static Failure validateBinaryComparison(BinaryScalarFunction bc) {
+        assert bc instanceof BinaryComparison || bc instanceof InsensitiveBinaryComparison;
```

Comment:
Do we want do to it now?
I implemented this validation consistently with other `BinaryComparison` operators, but I agree that it would be better to have the validation (and the resolution, that is a different aspect) inside the single operator classes.
IMHO it would be better to be consistent now, and then do a consistent refactoring for all the operators

Reply:
Ok, fixed. Since we are only accepting strings now, the checks are much easier. I just had to remove the Verifier logic